src/services/githubauth.service.js

```diff
@@ -124,6 +125,44 @@ export default class GithubAuthenticationService {
     return tokenCreated;
   }
 
+  /**
+   * First exchanges code and state for a github token.
+   * Then adds a github token for the given user into the githubTokens table. If a token already
+   * exists for the user, it will automatically update the old one.
+   * @param userId the user id to add the token for
+   * @param code the code to exchange for token
+   * @param state the state to verify identity
+   * @param username the github username of the user
+   * @returns {Object} the token that was created
+   */
+  async getAndSetGithubTokenForUser(userId, code, state, username) {
+    try {
+      const secrets = config.get('oauth_secrets');
+      const res = await fetch('https://cors-anywhere.herokuapp.com/https://github.com/login/oauth/access_token', {
+        method: 'POST',
+        body: JSON.stringify({
+          client_id: '6b57706945fa18ee0397',
```

Comment:
I think we should move the client_id to the config file. This way it easier to reuse across the application if need be and all information is one place.